### PR TITLE
KOGITO-4720: MongoDB persistence support for Job Service

### PR DESCRIPTION
**Pull request by xieshenzh**
Description: MongoDB persistence support for Job Service
Jira: https://issues.redhat.com/browse/KOGITO-4720

Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](https://github.com/kiegroup/kogito-runtimes#contributing-to-kogito)
- [x] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [x] Pull Request title contains the target branch if not targeting master: `[0.9.x] KOGITO-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>Pull Request</b>  
  Please add comment: <b>Jenkins retest this</b>
 
* <b>Quarkus LTS checks</b>  
  Please add comment: <b>Jenkins run LTS</b>

* <b>Native checks</b>  
  Please add comment: <b>Jenkins run native</b>

</details>

### CODE REVIEW

**Comment by xieshenzh** [comment]
Assembly:
https://github.com/kiegroup/kogito-runtimes/pull/1218
https://github.com/kiegroup/kogito-apps/pull/756

File: jobs-service/jobs-service-mongodb/pom.xml [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+                <executions>
+                    <execution>
+                        <goals>
+                            <goal>build</goal>
+                        </goals>
+                    </execution>
+                </executions>
+            </plugin>
+            <plugin>
+                <artifactId>maven-failsafe-plugin</artifactId>
```
Comment (cristianonicolai) [inline jobs-service/jobs-service-mongodb/pom.xml:92]: @xieshenzh can you review the pom, I have done some changes in the structure, please follow https://github.com/kiegroup/kogito-apps/blob/master/jobs-service/jobs-service-postgresql/pom.xml#L100

File: jobs-service/jobs-service-mongodb/src/main/resources/application.properties [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+quarkus.log.console.async=true
+##File
+quarkus.log.file.enable=true
+quarkus.log.file.format=%d{yyyy-MM-dd HH:mm:ss,SSS} %h %-5p [%c:%L] (%t) %s%e%n
+quarkus.log.file.path=jobs-service.log
+quarkus.log.file.async=true
+quarkus.log.file.rotation.file-suffix=yyyy-MM-dd
+
+#Container image
+quarkus.container-image.build=true
```
Comment (cristianonicolai) [inline jobs-service/jobs-service-mongodb/src/main/resources/application.properties]:
```suggestion
quarkus.container-image.build=${quarkus.build.image:true}
```

**Review by cristianonicolai — COMMENTED** [review]
@xieshenzh can you also add a Native test

**Comment by VishnuBharghavReddy** [comment]
Please approve ASAP.

**Comment by xieshenzh** [comment]
> @xieshenzh can you also add a Native test

@cristianonicolai do you mean removing the `native` tests from the pom? I saw the native tests are removed for postgresql and infinispan.

File: jobs-service/jobs-service-mongodb/pom.xml [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>org.kie.kogito</groupId>
+        <artifactId>jobs-service</artifactId>
+        <version>2.0.0-SNAPSHOT</version>
+    </parent>
+
+    <artifactId>jobs-service-mongodb</artifactId>
+    <name>Kogito :: Jobs Service MongoDB</name>
```
Comment (cristianonicolai) [inline jobs-service/jobs-service-mongodb/pom.xml]:
```suggestion
    <name>Kogito :: Jobs Service :: MongoDB</name>
```

File: jobs-service/jobs-service-mongodb/pom.xml [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+                        </configuration>
+                    </execution>
+                    <execution>
+                        <id>testWithKeycloak</id>
+                        <goals>
+                            <goal>integration-test</goal>
+                        </goals>
+                        <configuration>
+                            <includes>
+                                <include>**/MongoDBKeycloak*IT.java</include>
```
Comment (cristianonicolai) [inline jobs-service/jobs-service-mongodb/pom.xml]: to match the filters, please rename the class to `KeycloakMongoDB`

**Comment by cristianonicolai** [comment]
@xieshenzh correct, no need to set native profile, will be added automatically, just need to have native tests starting with Native*

**Review by cristianonicolai — APPROVED** [review]

**Comment by VishnuBharghavReddy** [comment]
Please share the image URL of jobs-service-mongodb-persistence.

**Comment by tiagodolphine** [comment]
I'll review this one today, @MarianMacik it would be great if you review this too.
Thanks

**Comment by cristianonicolai** [comment]
@tiagodolphine @MarianMacik looks good to you?

**Comment by VishnuBharghavReddy** [comment]
@tiagodolphine @MarianMacik Could you please review and close this ASAP.

**Review by tiagodolphine — APPROVED** [review]
LGTM, maybe it is worth adding a kind of docker-compose file to make things easier for users to get started, and docs + examples (like https://github.com/kiegroup/kogito-examples/tree/stable/process-timer-quarkus) could have some notes with the persistence possibilities for job-service.

**Comment by cristianonicolai** [comment]
@tiagodolphine agree, we need the docs to also highlight these options. @xieshenzh could you look into that on follow up pr?

**Comment by xieshenzh** [comment]
> @tiagodolphine agree, we need the docs to also highlight these options. @xieshenzh could you look into that on follow up pr?

@cristianonicolai @tiagodolphine please find the PRs for docs and timer examples below:
Docs PR: https://github.com/kiegroup/kie-docs/pull/3555
Timer example PR: https://github.com/kiegroup/kogito-examples/pull/683

**Pull request merged**